code/datums/components/storage/concrete/pockets.dm

```diff
@@ -17,16 +17,30 @@
 
 /datum/component/storage/concrete/pockets/small
 	max_items = 1
+	max_w_class = WEIGHT_CLASS_SMALL
 	attack_hand_interact = FALSE
 
 /datum/component/storage/concrete/pockets/tiny
 	max_items = 1
 	max_w_class = WEIGHT_CLASS_TINY
 	attack_hand_interact = FALSE
 
-/datum/component/storage/concrete/pockets/small/detective
+/datum/component/storage/concrete/pockets/small/detective/Initialize()
+	. = ..()
 	attack_hand_interact = TRUE // so the detectives would discover pockets in their hats
 
+	exception_hold = typecacheof(list(
+		/obj/item/katana, /obj/item/toy/katana, /obj/item/nullrod/claymore/katana,
+		/obj/item/gun/ballistic/automatic/tommygun
+		))
+
+/datum/component/storage/concrete/pockets/small/fedora/Initialize()
+	. = ..()
+	exception_hold = typecacheof(list(

```

Comment:
same

Reply:
this is not resolved, you need to be caching these typecache lists

Reply:
Resolving as mentioned to you on Discord, this line is the only time this specific typecache is created.

Reply:
This is still not resolved, the list needs to be created once the reused every time a new instance of this component is created

Reply:
Alright, I think it's **actually** resolved now.